### ISSUE TITLE
How to wait/fetch select('core').getMedia() response?

### ISSUE THREAD

**Issue by mr660** [WordPress/gutenberg#16003]
I'm trying to make component `ImageControl` where  using `MediaPlaceholder` and `MediaUpload`. I'm getting media using `select('core').getMedia()` but it has some delay that I can't process. How i can set state after i get Image? This code works but image is shown only after de-selecting block.

```
const wp = window.wp
const { select } = wp.data
const { withState } = wp.compose
const { Fragment } = wp.element
const { MediaPlaceholder, MediaUpload } = wp.editor
const { Spinner } = wp.components

const ALLOWED_MEDIA_TYPES = ['image']

function ImageControl (
  {
    value,
    onSelect,
    setState,
    isEditing
  }
) {
  let image

  if (typeof value !== 'undefined') {
    image = select('core').getMedia(value)
  }

  const onSelectValue = (media) => {
    setState({ isEditing: true })
    onSelect(media.id)
  }

  console.log(value, image)

  return (
    <Fragment>
      {(() => {
        if (!isEditing) {
          return (
            <MediaPlaceholder
              labels={{ title: 'Image' }}
              multiple={false}
              allowedTypes={ALLOWED_MEDIA_TYPES}
              onSelect={onSelectValue}
            />
          )
        } else {
          if (typeof image !== 'undefined') {
            return (
              <MediaUpload
                value={value}
                allowedTypes={ALLOWED_MEDIA_TYPES}
                onSelect={onSelectValue}
                render={({ open }) => (
                  <div className="components-image-control">
                    <button
                      className="components-image-control__btn"
                      onClick={open}
                      onSelect={onSelectValue}
                    >&nbsp;</button>
                    <img src={image.source_url}/>
                  </div>
                )}
              />
            )
          }
          return (
            <Spinner/>
          )
        }
      })()}
    </Fragment>
  )
}

export default withState({
  isEditing: false
})(
  ImageControl
)
``

**Comment by Soean** [comment]
Related to #16004 and #14623

**Comment by swissspidy** [comment]
Closing as a duplicate of #14623 because the same question has been asked there.
